Lib/test/test_types.py

```diff
@@ -864,6 +864,22 @@ def __prepare__(*args):
         self.assertIs(ns, expected_ns)
         self.assertEqual(len(kwds), 0)
 
+    def test_bad___prepare__(self):
+        # __prepare__() must return a mapping.
+        class BadMeta(type):
+            def __prepare__(*args):
+                return None
+        with self.assertRaises(TypeError):
```

Comment:
This should use `self.assertRaisesRegex` to check we're getting the error message we expect rather than a different TypeError ( like "`NoneType` is not subscriptable")